cwltool/job.py

```diff
@@ -347,14 +355,20 @@ def stderr_stdout_log_path(
                 processStatus = "permanentFail"
 
             if processStatus != "success":
-                if rcode < 0:
+                if runtimeContext.kill_switch.is_set():
+                    processStatus = "killed"
+                    return
```

Comment:
Instead of `return` here and then putting a huge amount of code in a `finally` block, how about turning this into `raise WorkflowKillSwitch()` or `raise WorkflowException("killed")` and then revert the cleanup code back the way it was?  You're trying to exit early but I think once the kill switch is set, all the other processes are going to shut down in a (more or less) orderly fashion so you should just let the existing code do its thing, unless there's a specific bad behavior you need to avoid.